### PR TITLE
Feature/bundle new

### PR DESCRIPTION
**Pull request by xpositivityx**
* Adds commands for bundle new
* Adds some testing niceties for working with the mock filesystem.
* Moves testing to the command level

### CODE REVIEW

File: internal/commands/generate_new_bundle_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+		TemplateName:   "terraform",
+		TemplateRepo:   "massdriver-cloud/infrastructure-templates",
+		TemplateSource: "/home/md-cloud",
+		Name:           "aws-dynamodb",
+		Access:         "private",
+		Description:    "whatever",
+		Connections: map[string]string{
+			"massdriver/aws-authentication": "auth",
+		},
+		CloudPrefix:     "aws",
```
Comment (coryodaniel) [inline internal/commands/generate_new_bundle_test.go]: Nit: CloudAbbreviation? tell them what it is not how to use this? I regret "name_prefix" in MD metadata ... its the identifier for the bundle which may be used as a prefix but now that cursed named is everywhere.

File: internal/commands/generate_new_bundle_test.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+	checkErr(err, t)
+
+	wantConnections := map[string]interface{}{
+		"properties": map[string]interface{}{
+			"massdriver/aws-authentication": map[string]interface{}{
+				"$ref": "auth",
+			},
+		},
+		"required": []interface{}{"massdriver/aws-authentication"},
```
Comment (coryodaniel) [inline internal/commands/generate_new_bundle_test.go:129]: I dont follow how this is true given the mock func below.

File: internal/commands/generate_new_bundle_test.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	}
+
+	templateData := mockTemplateData(writePath)
+
+	err = commands.GenerateNewBundle(bundleCache, templateData)
+
+	checkErr(err, t)
+
+	wantTopLevel := []string{
+		"massdriver.yaml",
```
Comment (coryodaniel) [inline internal/commands/generate_new_bundle_test.go:76]: I dont follow how this is true given the mock func below. Where do we have a mock of the massdriver.yaml?

File: internal/mockfilesystem/main.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	templatePath := "terraform"
+	srcPath := "src"
+
+	directories := []string{
+		path.Join(rootTemplateDir, repoPath),
+		path.Join(rootTemplateDir, repoPath, templatePath),
+		path.Join(rootTemplateDir, repoPath, templatePath, srcPath),
+	}
+
+	massdriverYamlTemplate, err := os.ReadFile("../templatecache/testdata/massdriver.yaml.txt")
```
Comment (xpositivityx) [inline internal/mockfilesystem/main.go:32]: @coryodaniel The SetupBundle function here writes a massdriver.yaml template file to the memory FS.

**Pull request merged**